src/frequenz/client/microgrid/component/_types.py

```diff
@@ -0,0 +1,65 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""All known component types."""
+
+from typing import TypeAlias
+
+from ._battery import BatteryTypes, UnrecognizedBattery, UnspecifiedBattery
+from ._chp import Chp
+from ._converter import Converter
+from ._crypto_miner import CryptoMiner
+from ._electrolyzer import Electrolyzer
+from ._ev_charger import EvChargerTypes, UnrecognizedEvCharger, UnspecifiedEvCharger
+from ._fuse import Fuse
+from ._grid_connection_point import GridConnectionPoint
+from ._hvac import Hvac
+from ._inverter import InverterTypes, UnrecognizedInverter, UnspecifiedInverter
+from ._meter import Meter
+from ._precharger import Precharger
+from ._problematic import (
+    MismatchedCategoryComponent,
+    UnrecognizedComponent,
+    UnspecifiedComponent,
+)
+from ._relay import Relay
+from ._voltage_transformer import VoltageTransformer
+
+UnspecifiedComponentTypes: TypeAlias = (
+    UnspecifiedBattery
+    | UnspecifiedComponent
+    | UnspecifiedEvCharger
+    | UnspecifiedInverter
+)
+"""All unspecified component types."""
+
+UnrecognizedComponentTypes: TypeAlias = (
+    UnrecognizedBattery
+    | UnrecognizedComponent
+    | UnrecognizedEvCharger
+    | UnrecognizedInverter
+)
+
+ProblematicComponentTypes: TypeAlias = (
+    MismatchedCategoryComponent | UnrecognizedComponentTypes | UnspecifiedComponentTypes
+)
+"""All possible component types that has a problem."""
+
+ComponentTypes: TypeAlias = (
```

Comment:
Maybe rename `Component` -> `AbstractComponent` and this to `Component`?

These things are not lists or iterables, so they can't have plural names.

Reply:
Resolving this to keep the discussion in one thread, but whatever solution we agree on will be done for all `*Types` aliases.